src/node/in_memory_ext.rs

```diff
@@ -0,0 +1,321 @@
+            .and_then(|mut writer| {
+                let num_blocks = num_blocks.unwrap_or_else(|| U64::from(1));
+                let interval_ms = interval
+                    .unwrap_or_else(|| U64::from(1))
+                    .saturating_mul(1_000.into());
+                if num_blocks.is_zero() {
+                    return Err(anyhow!(
+                        "Number of blocks must be greater than 0".to_string(),
+                    ));
+                }
+                utils::mine_empty_blocks(&mut writer, num_blocks.as_u64(), interval_ms.as_u64());
+                tracing::info!("👷 Mined {} blocks", num_blocks);
+
+                Ok(true)
+            })
+    }
+
+    pub fn impersonate_account(&self, address: Address) -> Result<bool> {
+        self.get_inner()
+            .write()
+            .map_err(|err| anyhow!("failed acquiring lock: {:?}", err))
+            .map(|mut writer| {
+                if writer.set_impersonated_account(address) {
+                    tracing::info!("🕵️ Account {:?} has been impersonated", address);
+                    true
+                } else {
+                    tracing::info!("🕵️ Account {:?} was already impersonated", address);
+                    false
+                }
+            })
+    }
+
+    pub fn stop_impersonating_account(&self, address: Address) -> Result<bool> {
+        self.get_inner()
+            .write()
+            .map_err(|err| anyhow!("failed acquiring lock: {:?}", err))
+            .map(|mut writer| {
+                if writer.stop_impersonating_account(address) {
+                    tracing::info!("🕵️ Stopped impersonating account {:?}", address);
+                    true
+                } else {
+                    tracing::info!(
+                        "🕵️ Account {:?} was not impersonated, nothing to stop",
+                        address
+                    );
+                    false
+                }
+            })
+    }
+
+    pub fn set_code(&self, address: Address, code: Vec<u8>) -> Result<()> {
+        self.get_inner()
+            .write()
+            .map_err(|err| anyhow!("failed acquiring lock: {:?}", err))
+            .map(|mut writer| {
+                let code_key = get_code_key(&address);
+                tracing::info!("set code for address {address:#x}");
+                let (hash, code) = bytecode_to_factory_dep(code);
+                let hash = u256_to_h256(hash);
+                writer.fork_storage.store_factory_dep(
+                    hash,
+                    code.iter()
+                        .flat_map(|entry| {
+                            let mut bytes = vec![0u8; 32];
+                            entry.to_big_endian(&mut bytes);
+                            bytes.to_vec()
+                        })
+                        .collect(),
+                );
+                writer.fork_storage.set_value(code_key, hash);
+            })
+    }
+}
```

Comment:
Should we also be moving the unit tests into this file? Feels like a majority of the business logic is here